linea-it/dri
# Integrate with DESaccess

A new integration with the DESaccess service was implemented.

- A new DesAccessApi class was developed, this class provides the necessary methods to interact with the Cutout and Tile Finder functions.
- A new workflow was created for the Cutout process submitted by the target viewer.
- Endpoints were created that return the list of image files and catalogs for a tile.
- Some corrections were made in the classes of access to Catalogs.
- Some libraries needed to be updated.
- The models that manage the cutout jobs have been changed.
- The configuration variables for integration with the old Descut have been removed and a new config has been created for DESaccess.

How to test:
- Before switching to this branch, delete all records at admin-> product-> cutoutjob.
- change local_vars.py replacing the DESCUTOUT_SERVICE variable with the new DESACCESS_API configuration an example is in local_vars.template
- switch to this brach, and build the backend image.

To test the Cutout workflow:
- A test method is available in dri/api/test. it is necessary to have a list of targets registered and the id of that product must be in the test function.
- follow the log/backend/cutout.log
- Check by admin the models admin/product/cutoutjob, desjob, cutout.

To test the Tile Finder endpoints:
- Returns all files to a dataset, considering only its release: http://localhost/dri/api/dataset/51148/desprcess_tile_info/

- Returns all files for a tile: http://localhost/dri/api/tiles/97264/desprcess_tile_info/

Test the method that generates a Download URL for a file:
- Choose one of the urls returned by the previous methods, make a POST request to the address: http://localhost/dri/api/dataset/desaccess_get_download_url  passing the file_url parameter.
